### ISSUE TITLE
PR description using commit messages may produce wrong results

### ISSUE THREAD

**Issue by zmeir** [Codium-ai/pr-agent#179]
I've noticed that when calling the `/describe` command, the prompt uses the commit message in the PR to generate the PR description. In cases where a commit is no longer relevant (for example - it was later reverted in the same PR) this produces the wrong output in the PR description.

See this small PR example:  
* https://github.com/zmeir/pr-agent/pull/2

In this example I've made 2 commit:
1. The first commit updates the docstring of some function
2. The second commit reverts the docstring to its original state and updates the max number of tokens in another file

After these 2 commits I called the `/describe` command, and as you can see, the PR title and description say that this PR includes changes to the docstring, but if you look at the diff with the main branch you can clearly see this is not true.

**Comment by mrT23** [comment]
interesting

My intuition was that the commit messages might direct the LLM to focus on the most important changes when choosing the title.

I will try tweaking the prompt a bit, to make it more conservative in using the commit messages

**Comment by zmeir** [comment]
I get that, but I think there are 2 issues with how the LLM might interpret the commit messages that are counter-productive:
1. The commit messages are presented as just a plain list of messages, so it might interpret them as "statements that are true about this PR", But in reality, this list of commits is a lineage of incremental changes that apply one on top of the other. Think of these 2 commits: "added file a" and "deleted file a". If you look at just those 2 messages without understanding the lineage it may seem that 2 separate things happened - a file addition and a file deletion, and you will probably see those 2 things in the description. But in reality, nothing really changed.
2. It's also important to remember that when people develop on their branch the commit messages aren't always that informative while they're debugging and testing their code. There's probably some heuristic that can help determine if the commit message is meaningful and relevant to what actually happened in the commit, and not just one of many dummy comments like "fixed something" that usually happen during debugging cycles.

**Comment by mrT23** [comment]
i agree with 1 and 2. i want to refine the prompt to address that ('Notice that commit message may be partial, simplistic, or non-informative..')

**Comment by mrT23** [comment]
p.s.
We tried sweepAI on your issue. These are the results:

https://github.com/mrT23/pr-agent/issues/2

https://github.com/mrT23/pr-agent/pull/4/files

among the 3 issues we tried, I think this was the best result (although I am aiming for a different solution)

**Comment by zmeir** [comment]
@mrT23 hey that's pretty cool! But I think the solution offered by sweep is a bit too simplistic as it only addresses the issue of perfectly reverted commits. I think that, as you mentioned, tuning the prompt is the way to go.

**Comment by mrT23** [comment]
see https://github.com/Codium-ai/pr-agent/pull/188
